### PR TITLE
Create faucet chat based bot

### PR DESCRIPTION
**Pull request by StringNick**
#### Project Abstract

   Sybil-resistant faucet is a generic chat bot based faucet solution that can be used on any existing parachain (substrate-? 
   based chain, either pallets or ink! smart contracts). 

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [x] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (bank details via email _or_ BTC, Ethereum (USDC/DAI) or Polkadot/Kusama (USDT) address in the application).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the application. Looks interesting. I have two initial questions: 
- What exactly do you mean by "I will create a module"? It would be nice to have more technical details for each delivery, such as the programming language.
- Could you also integrate the ink! or pallet integration into the milestone table? The milestone table lists the requirements of our contracts.

**Comment by StringNick** [comment]
@Noc2 
Thank for review
> * What exactly do you mean by "I will create a module"? It would be nice to have more technical details for each delivery, such as the programming language.
Done
> * Could you also integrate the ink! or pallet integration into the milestone table? The milestone table lists the requirements of our contracts.
I added substrate demo row in table, but i dont exactly understand, what you mean. Because bot will use substrate rpc golang sdk to send, using balance pallet methods.

**Review by Noc2 — APPROVED** [review]
Thanks for the update and the additional information. Could you also integrate the substrate RPC Golang SDK into the milestone table? Simply saying that it will be based on it and maybe even linking to the GitHub code. I ask for something like this mostly to make sure that you will support Substrate with your delivery and not any other potential chain. Independent of it, I'm already happy to go ahead with it and mark it as ready for review.

**Comment by StringNick** [comment]
> Thanks for the update and the additional information. Could you also integrate the substrate RPC Golang SDK into the milestone table? Simply saying that it will be based on it and maybe even linking to the GitHub code. I ask for something like this mostly to make sure that you will support Substrate with your delivery and not any other potential chain. Independent of it, I'm already happy to go ahead with it and mark it as ready for review.

i added link to grpc substrate library for golang, i suppose this library in ur open-source wiki projects

**Review by Noc2 — APPROVED** [review]
Thanks

**Review by takahser — CHANGES_REQUESTED** [review]
@StringNick, I noticed that you mentioned [two similar projects](https://github.com/w3f/Grants-Program/pull/1758/files?diff=unified&w=0#diff-4694476b85dfbcea526e7c67cc0e3de7493af9a595e7eb7e936731ce47b4c849R47-R49), but it would be beneficial if you could further elaborate on how your solution differs from them. It appears that the existing projects employ Element and Telegram to prevent spamming, whereas I believe your solution aims to be more generic in terms of sybil-resistance. It would be good if you could provide additional details in your proposal, ensuring clarity about the unique value your solution brings.

**Comment by StringNick** [comment]
> @StringNick, I noticed that you mentioned [two similar projects](https://github.com/w3f/Grants-Program/pull/1758/files?diff=unified&w=0#diff-4694476b85dfbcea526e7c67cc0e3de7493af9a595e7eb7e936731ce47b4c849R47-R49), but it would be beneficial if you could further elaborate on how your solution differs from them. It appears that the existing projects employ Element and Telegram to prevent spamming, whereas I believe your solution aims to be more generic in terms of sybil-resistance. It would be good if you could provide additional details in your proposal, ensuring clarity about the unique value your solution brings.

Done

**Review by takahser — DISMISSED** [review]
@StringNick by "platform" I assume you mostly mean messaging platforms like Discord, Element, Telegram, etc. Maybe just add some example platforms, to make it clear what you mean by "platform". Apart from that, I'm happy to support your proposal.

**Comment by StringNick** [comment]
@takahser added, ty for review.

**Review by Noc2 — DISMISSED** [review]

File: applications/faucet-bot.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+### Ecosystem Fit
+
+Many dApps are facing an issue where it’s difficult to onboard new users. Thus, the goal is to simplify the process by making it easier for parachain and dApp developers to spin up their own faucets, and give users free tokens without people exploiting the system. In order to make the system sybil-resistant, centralised solutions like Discord, Matrix and any other chat based platforms, that will uniquely identify users, and enable requesting tokens to the account only once per given time period.
+
+Some similar projects include:
+- https://github.com/paritytech/substrate-matrix-faucet
+- https://github.com/OAXFoundation/substrate-faucet
+
+The advantages of this project are that similar projects are implemented specifically for one platform, this project will be implemented on the interface, and the platforms(Discord, matrix and etc) will be like addons (classes) that correspond to the interface, which will allow you to easily connect new platforms, do not depend on the implementation of platforms, will allow you to run unlimited number of platforms at the same time.
```
Comment (takahser) [inline applications/faucet-bot.md:51]:
```suggestion
The advantages of this project are that similar projects are implemented specifically for one platform, this project will be implemented on the interface, and the platforms (Discord, matrix and etc) will be like addons (classes) that correspond to the interface, which will allow you to easily connect new platforms, do not depend on the implementation of platforms, will allow you to run unlimited number of platforms at the same time.
```

**Review by takahser — APPROVED** [review]
@StringNick LGTM, thanks for the update

File: applications/faucet-bot.md (hunk 1 above)
Comment (takahser) [reply applications/faucet-bot.md]: just fixed a space here

**Review by Noc2 — APPROVED** [review]

**Pull request merged**